src/Adapter/Product/Combination/Repository/CombinationRepository.php

```diff
@@ -366,4 +368,24 @@ public function getCombinationIdsByAttributes(ProductId $productId, array $attri
             return new CombinationId((int) $combination['id_product_attribute']);
         }, $result);
     }
+
+    public function updateCombinationStock(ProductId $productId, OutOfStockType $outOfStockType, ShopConstraint $shopConstraint = null): void
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->update(sprintf('%sstock_available', $this->dbPrefix), 'ps')
+            ->set('ps.out_of_stock', (string) $outOfStockType->getValue())
+            ->where('ps.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+        ;
+
+        if ($shopConstraint !== null && $shopConstraint->getShopId() !== null) {
```

Comment:
checking strictly against null doesn't add any value, because arguments are already strictly typed, so i IMO its cleaner to read without `null` checks:

```suggestion
        if ($shopConstraint && $shopConstraint->getShopId()) {
```

Reply:
Also shouldn't we handle shopGroupId?
Lets say if $shop->constraint is ShopGroupId, then we need to check `where id_shop IN (:shopIds)` (shop ids being the shops in group)?

Reply:
You are right about shopGroupIds, I've aded code to handle them. Considering the 'short if' notation, I prefer keep it that way because I don't like the `if($object)` notation and I prefer testing boolean results.